lessonbuilder/components/src/java/org/sakaiproject/lessonbuildertool/model/SimplePageToolDaoImpl.java

```diff
@@ -894,7 +894,7 @@ public boolean isPageVisited(long pageId, String userId, String owner) {
 
 	    Object [] fields = new Object[2];
 	    fields[0] = Long.toString(pageId);
-	    fields[1] = userId;
+	    fields[1] = (userId == null)?".anon":userId;
```

Comment:
The call from `SimplePageBean.addComment()` passes in an empty string, so either this check should also look for an empty string or the calling code should pass in `null` instead.